src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -42,11 +42,14 @@ CustomTextLayout::CustomTextLayout(gsl::not_null<IDWriteFactory1*> const factory
     _localeName.resize(gsl::narrow_cast<size_t>(format->GetLocaleNameLength()) + 1); // +1 for null
     THROW_IF_FAILED(format->GetLocaleName(_localeName.data(), gsl::narrow<UINT32>(_localeName.size())));
 
-    for (const auto& cluster : clusters)
+    RenderClusterIterator it = clusterIter;

```

Comment:
Because we are using `RenderClusterIterator const`, the copy is needed. It could be `RenderClusterIterator&` and then it can be modifed by the caller. But that's just kind of implicit to me. I prefer the copying.

Reply:
Nah, I like what you did. It's constant for the specification of the custom layout, but then the internals of the layout makes the copy to walk through it.